docs/design-doc-provider-identity-based-auth.md

```diff
@@ -0,0 +1,571 @@
+# Identity Based Authentication for Crossplane Providers
+
+* Owner: Alper Rifat Uluçınar (@ulucinar)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Background
+Crossplane providers need to authenticate themselves to their respective Cloud
+providers. This establishes an identity for the Crossplane provider that's later
+used by the Cloud provider to authorize the requests made by the Crossplane
+provider and for various other purposes such as audit logging, etc. Each
+Crossplane provider supports a subset of the underlying Cloud provider's
+authentication mechanisms and this subset is currently implemented in-tree,
+i.e., in the Crossplane provider's repo, there exists a CRD that's
+conventionally named as `ProviderConfig` and each managed resource of the
+provider has a [v1.Reference][v1.reference] to a `ProviderConfig` CR. This
+`ProviderConfig` holds the authentication configuration (chosen authentication method,
+any required credentials for that method, etc.) together with any other provider
+specific configuration. Different authentication methods and/or different sets
+of credentials can be configured using separate cluster-scoped `ProviderConfig`
+CRs and by having different managed resources refer to these `ProviderConfig`
+instances. 
+
+The Crossplane provider establishes an identity for the requests it will issue
+to the Cloud provider in the [managed.ExternalConnecter][managed.ExternalConnecter]'s `Connect`
```

Comment:
The `[managed.ExternalConnecter]` link is left as TODO and opens a 404 error page, I think it would be good to fix this.

Reply:
Unfortunately can't find document link to resolve this.

Reply:
We can use the following here: https://pkg.go.dev/github.com/crossplane/crossplane-runtime@v0.19.2/pkg/reconciler/managed#ExternalConnecter

Reply:
Fixed